semgrep-core/src/optimizing/Analyze_rule.ml

```diff
@@ -517,8 +522,14 @@ let regexp_prefilter_of_rule r =
   let id, t = r.R.id in
   let k = PI.file_of_info t ^ "." ^ id in
   Common.memoized hmemo k (fun () ->
-      match r.mode with
-      | R.Search pf ->
-          let f = R.formula_of_pformula pf in
-          regexp_prefilter_of_formula f
-      | R.Taint spec -> regexp_prefilter_of_taint_rule t spec)
+      try
+        match r.mode with
+        | R.Search pf ->
+            let f = R.formula_of_pformula pf in
+            regexp_prefilter_of_formula f
+        | R.Taint spec -> regexp_prefilter_of_taint_rule t spec
+      with
+      (* TODO: see tests/OTHER/rules/tainted-filename.yaml *)
+      | Stack_overflow ->
```

Comment:
`regexp_prefilter_of_taint_rule` already has a top-level `try-with`, maybe one could just add a case for `| Stack_overflow ->` there?

Reply:
yes, good idea.

Reply:
hmm but then I don't have the rule id to logger the issue
simpler to keep there then.